### DISCUSSION TITLE
Reproduction of example fails. No raster is shown

### DISCUSSION THREAD

**Discussion by lwlR** [banesullivan/localtileserver#73] [Q&A] [answered]
Thank you for providing this helpful package! I would like to reproduce your example with my dataset. Unfortunately, the raster is not shown at all (OSM basemap is). I am working on Jupyterhub with Python 3.8.10:

```
from localtileserver import get_leaflet_tile_layer, TileClient
from ipyleaflet import Map

tile_client = TileClient('20180713_025515_all_tiles_merc.tif')

t = get_leaflet_tile_layer(tile_client)

m = Map(center=tile_client.center(), zoom=6)
m.add_layer(t)
m
```

Gdalinfo gives:
```

Driver: GTiff/GeoTIFF
Files: 20180713_025515_all_tiles_merc.tif
Size is 17664, 47616
Coordinate System is:
PROJCRS["WGS 84 / Pseudo-Mercator",
    BASEGEOGCRS["WGS 84",
        DATUM["World Geodetic System 1984",
            ELLIPSOID["WGS 84",6378137,298.257223563,
                LENGTHUNIT["metre",1]]],
        PRIMEM["Greenwich",0,
            ANGLEUNIT["degree",0.0174532925199433]],
        ID["EPSG",4326]],
    CONVERSION["unnamed",
        METHOD["Popular Visualisation Pseudo Mercator",
            ID["EPSG",1024]],
        PARAMETER["Latitude of natural origin",0,
            ANGLEUNIT["degree",0.0174532925199433],
            ID["EPSG",8801]],
        PARAMETER["Longitude of natural origin",0,
            ANGLEUNIT["degree",0.0174532925199433],
            ID["EPSG",8802]],
        PARAMETER["False easting",0,
            LENGTHUNIT["metre",1],
            ID["EPSG",8806]],
        PARAMETER["False northing",0,
            LENGTHUNIT["metre",1],
            ID["EPSG",8807]]],
    CS[Cartesian,2],
        AXIS["easting",east,
            ORDER[1],
            LENGTHUNIT["metre",1,
                ID["EPSG",9001]]],
        AXIS["northing",north,
            ORDER[2],
            LENGTHUNIT["metre",1,
                ID["EPSG",9001]]]]
Data axis to CRS axis mapping: 1,2
Origin = (12452509.151994634419680,6090502.413762843236327)
Pixel Size = (4.777314267823508,-4.777314267823508)
Metadata:
  AREA_OR_POINT=Area
Image Structure Metadata:
  COMPRESSION=LZW
  INTERLEAVE=PIXEL
  LAYOUT=COG
Corner Coordinates:
Upper Left  (12452509.152, 6090502.414) (111d51'46.05"E, 47d54' 5.81"N)
Lower Left  (12452509.152, 5863025.818) (111d51'46.05"E, 46d30'48.66"N)
Upper Right (12536895.631, 6090502.414) (112d37'15.06"E, 47d54' 5.81"N)
Lower Right (12536895.631, 5863025.818) (112d37'15.06"E, 46d30'48.66"N)
Center      (12494702.392, 5976764.116) (112d14'30.56"E, 47d12'43.58"N)
Band 1 Block=512x512 Type=UInt16, ColorInterp=Red
  Overviews: 8832x23808, 4416x11904, 2208x5952, 1104x2976, 552x1488, 276x744, 138x372, 69x186
Band 2 Block=512x512 Type=UInt16, ColorInterp=Green
  Overviews: 8832x23808, 4416x11904, 2208x5952, 1104x2976, 552x1488, 276x744, 138x372, 69x186
Band 3 Block=512x512 Type=UInt16, ColorInterp=Blue
  Overviews: 8832x23808, 4416x11904, 2208x5952, 1104x2976, 552x1488, 276x744, 138x372, 69x186
Band 4 Block=512x512 Type=UInt16, ColorInterp=Undefined
  Overviews: 8832x23808, 4416x11904, 2208x5952, 1104x2976, 552x1488, 276x744, 138x372, 69x186
Band 5 Block=512x512 Type=UInt16, ColorInterp=Alpha
  Overviews: 8832x23808, 4416x11904, 2208x5952, 1104x2976, 552x1488, 276x744, 138x372, 69x186
```
Any idea, what I can try to solve this? Thank you!

**Comment by banesullivan** [comment]
That raster's gdalinfo looks good to me!

If no errors arise and the raster simply isn't appearing with that code, then I suspect you must be running in some sort of remote Jupyter environment, is that right?

This is a common scenario and there are a few threads addressing this topic. I have started some basic documentation to help with this use case here: https://localtileserver.banesullivan.com/installation/remote-jupyter.html

Additionally, the following threads might have some helpful context:

- https://github.com/banesullivan/localtileserver/issues/72#issuecomment-1082494133
- #29 
- #66

And there are some notes here: https://github.com/banesullivan/localtileserver/releases/tag/0.4.2

**Comment by lwlR** [comment, marked as answer]
Thank you for your quick response! With the links, I could solve the issue. 
Indeed, I am working on a remote computer with Jupyterhub.
For all, who also might have this problem, I share my steps which were 
necessary for me:
Follow instructions on the Remote Jupyter webpage:
https://localtileserver.banesullivan.com/installation/remote-jupyter.html

Update Node.js

Install JupyterLab extension
jupyter labextension install @jupyterlab/server-proxy 

Restart Jupyterhub
Am Mittwoch, 30. März 2022, 18:41:12 CEST schrieb Bane Sullivan:
> That raster's gdalinfo looks good to me!
> 
> If no errors arise and the raster simply isn't appearing with that code,
> then I suspect you must be running in some sort of remote Jupyter
> environment, is that right?
> 
> This is a common scenario and there are a few threads addressing this topic.
> I have started some basic documentation to help with this use case here:
> https://localtileserver.banesullivan.com/installation/remote-jupyter.html
> 
> Additionally, the following threads might have some helpful context:
> 
> -
> https://github.com/banesullivan/localtileserver/issues/72#issuecomment-1082
> 494133 - #29
> - #66
> 
> And there are some notes here:
> https://github.com/banesullivan/localtileserver/releases/tag/0.4.2

**Reply by banesullivan** [reply]
Thank you for following up with the steps to resolve this issue
